Add Creases plugin

liamcain opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:
https://github.com/liamcain/obsidian-creases

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

lishid commented:
https://github.com/liamcain/obsidian-creases/blob/09ae6194032e629a63f10c7ac7a9a94697f80d90/src/index.ts#L38
Editor commands should always check regardless of whether it's checking or not, to avoid race conditions where your function is called after the checked condition has changed.

Example of how it should be handled: https://marcus.se.net/obsidian-plugin-docs/guides/commands#editor-commands

https://github.com/liamcain/obsidian-creases/blob/09ae6194032e629a63f10c7ac7a9a94697f80d90/src/index.ts#L70
Same as above.

https://github.com/liamcain/obsidian-creases/blob/09ae6194032e629a63f10c7ac7a9a94697f80d90/src/index.ts#L124
Missing null check on view.

https://github.com/liamcain/obsidian-creases/blob/09ae6194032e629a63f10c7ac7a9a94697f80d90/src/index.ts#L285
Same here.

liamcain commented:
> Editor commands should always check regardless of whether it's checking or not, to avoid race conditions where your function is called after the checked condition has changed

Aah, good point. Fixed!

And added null checks for `view`. I was confused why Typescript let me get away with that but silly me, I didn't have `strictNullChecks` enabled on my project _tsk tsk_.

---

Thanks for the review and for looking the other way at all the hacks 🙈
